src/scripts/main.js

```diff
@@ -1,3 +1,34 @@
 'use strict';
 
 // write code here
+document.addEventListener('DOMContentLoaded', () => {
+  const headers = document.querySelectorAll('th');
+  const tableBody = document.querySelector('tbody');
+
+  const rows = Array.from(tableBody.querySelectorAll('tr'));
+
+  headers.forEach((header, index) => {
+    header.addEventListener('click', () => {
+      const sortedRows = [...rows].sort((a, b) => {
+        const aText = a.cells[index].textContent.trim();
+        const bText = b.cells[index].textContent.trim();
+
+        if (
+          !isNaN(parseFloat(aText.replace(/[^0-9.-]+/g, ''))) &&
+          !isNaN(parseFloat(bText.replace(/[^0-9.-]+/g, '')))
+        ) {
+          const aValue = parseFloat(aText.replace(/[^0-9.-]+/g, ''));
+          const bValue = parseFloat(bText.replace(/[^0-9.-]+/g, ''));
```

Comment:
You are using `parseFloat(aText.replace(/[^0-9.-]+/g, ''))` twice for each cell content. This is unnecessary repetition and could affect performance for larger tables. Consider storing the result in a variable and reusing it.